File: elixir/rna-transcription/rna_transcription.exs
defmodule RNATranscription do
  @paired_rna %{?A => ?U, ?C => ?G, ?T => ?A, ?G => ?C}

  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

  iex> RNATranscription.to_rna('ACTG')
  'UGAC'
  """
  @spec to_rna([char]) :: [char]
  def to_rna([]), do: []
  def to_rna([head | rest]), do: [@paired_rna[head] | to_rna(rest)]
end

File: python/rna-transcription/rna_transcription.py
dic = {'G': 'C', 'C': 'G', 'T': 'A', 'A': 'U'}

def to_rna(dna):
    rna = ''.join([dic.get(c, '') for c in dna if c in dic])

    if len(rna) == len(dna):
        return rna
    else:
        return ''
